php/example_code/iam/iam_basics/GettingStartedWithIAM.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Purpose
+ *
+ * Shows how to use AWS SDK for PHP v3 to get started using AWS Identity and Access Management (IAM).
+ * Create an IAM user, assume a role, and perform AWS actions.
+ *
+ * 1. Create a user that has no permissions.
+ * 2. Create a role and policy that grant s3:ListAllMyBuckets permission.
+ * 3. Grant the user permission to assume the role.
+ * 4. Create an S3 client object as the user and try to list buckets (this should fail).
+ * 5. Get temporary credentials by assuming the role.
+ * 6. Create an S3 client object with the temporary credentials and list the buckets (this should succeed).
+ * 7. Delete all the resources.
+ *
+ * You will need to run the following command to install dependencies:
+ * composer install
+ *
+ * Then run the example either directly with:
+ * php GettingStartedWithIAM.php
+ *
+ * or as a PHPUnit test:
```

Comment:
Alternatively, you can run the example as a PHPUnit test using the following command: